src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -521,16 +587,12 @@ private void generateDuplicateIndexes(final boolean useBarcodes, final boolean i
         // First just do the pairs
         log.info("Traversing read pair information and detecting duplicates.");
         for (final ReadEndsForMarkDuplicates next : this.pairSort) {
-            if (firstOfNextChunk == null) {
```

Comment:
I think that the reason is that it will equal the firstOfNextChunk for the last chunk of pairs, which should be different than the first fragment...there might be an edge case...so I'll add the nullify.